src/olympia/reviewers/tests/test_utils.py

```diff
@@ -3738,7 +3738,10 @@ def _test_request_legal_review(self, *, data=None):
             json={'job_id': '5678'},
             status=201,
         )
-
+        NeedsHumanReview.objects.create(
```

Comment:
Might be overkill but my intuition is that this test is missing something on two ends.

1. You aren't checking the the specific instance you created above is deleted, and we rely on implicit default values that is_active is even true. That's a possible false positive
2. the function that you're testing has a more subtly complex query embedded. It won't clear certain types of reasons. That's a false negative.

I almost want to see you create 2 instances, the one we expect to be removed and the one we expect not to be removed and then verify those specific instances are deleted..

https://github.com/mozilla/addons-server/blob/97c56fdb0bb2372b9f6ad26e99d3cb6d4bd0c6ec/src/olympia/reviewers/utils.py#L1040

Reply:
https://github.com/mozilla/addons-server/pull/23024/commits/95f23f427264acb6c910fda039dda052c5bae6f2